Flase positive on logcollector integration tests running in macOS systems.

This issue will be used to track false positive errors detected on logcollector integration tests on macOS systems.

## Error in test_location_custom_sockets
```
00:51:04.133  test_logcollector/test_location_custom_sockets/test_location_custom_sockets.py::test_location_custom_sockets_offline[target_custom_socket_mode_udp-batch_10000] FAILED [ 82%]Restarting wazuh-agent (via systemctl):  [  OK  ]

00:51:04.134  E                   TimeoutError: The expected command monitoring log has not been produced
```
https://devel.ci.wazuh.info/view/Tests/job/Test_integration/750/console

## Error in test_only_future_events

```
00:25:37.124  test_logcollector/test_configuration/test_basic_configuration_only_future_events.py::test_only_future_events[macos_no_20B] FAILED [ 33%]Mon Jun 21 06:27:20 UTC 2021

```

## 06/23/21

Today we've been investigating the issue with custom socket tests on macOS agents

```
=================================== FAILURES ===================================
00:34:28.234  _ test_location_custom_sockets_offline[target_custom_socket_mode_tcp-batch_500] _
00:34:28.234  
00:34:28.234  get_local_internal_options = {'logcollector.debug': 2, 'logcollector.queue_size': 2048, 'logcollector.state_interval': 5, 'monitord.rotate_log': 0}
00:34:28.234  configure_local_internal_options = None
00:34:28.234  get_configuration = {'apply_to_modules': ['test_location_custom_sockets'], 'metadata': {'location': '/tmp/wazuh-testing/test.log', 'log_fo...syslog'}}, {'target': {'value': 'custom_socket'}}], 'section': 'localfile'}], 'tags': ['test_location_custom_sockets']}
00:34:28.234  configure_environment = None, create_file_structure_module = None, batch = 500
00:34:28.234  create_socket = None, restart_logcollector = None
00:34:28.234  
00:34:28.234      @pytest.mark.parametrize("batch", batch_size, ids=[f"batch_{x}" for x in batch_size])
00:34:28.234      def test_location_custom_sockets_offline(get_local_internal_options, configure_local_internal_options,
00:34:28.234                                               get_configuration, configure_environment, create_file_structure_module,
00:34:28.234                                               batch, create_socket, restart_logcollector):
00:34:28.234          """Verify that event drops occur when the socket to which they are sent becomes unavailable.
00:34:28.234      
00:34:28.234          To do this logcollector is configured to forward events to a socket, and when the connection
00:34:28.234          has been established one event is written to the log file to force logcollector to connect
00:34:28.234          to the socket, then the socket is closed and batch of events is written to the log file,
00:34:28.234          in which case the event drops should be detected.
00:34:28.234      
00:34:28.234          Args:
00:34:28.234              get_local_internal_options (fixture): Get internal configuration.
00:34:28.234              configure_local_internal_options (fixture): Set internal configuration for testing.
00:34:28.234              get_configuration (fixture): Get configurations from the module.
00:34:28.234              configure_environment (fixture): Configure a custom environment for testing.
00:34:28.234              create_file_structure_module (fixture): Module scope version of create_file_structure.
00:34:28.234              batch (fixture): Event batches to be added to the test log file.
00:34:28.234              create_socket (fixture): Create a UNIX named socket for testing.
00:34:28.234              restart_logcollector (fixture): Reset log file and start a new monitor.
00:34:28.234          """
00:34:28.234          config = get_configuration['metadata']
00:34:28.234          global test_socket
00:34:28.234      
00:34:28.234          # Ensure that the log file is being analyzed
00:34:28.234          callback_message = lg.callback_analyzing_file(file=config['location'])
00:34:28.234          wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
00:34:28.234                                  error_message=lg.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING,
00:34:28.234                                  callback=callback_message)
00:34:28.234      
00:34:28.234          # Add one event to force logcollector to connect to the socket
00:34:28.234          with open(config['location'], 'a') as f:
00:34:28.234              f.write(f"{config['log_line']}\n")
00:34:28.234      
00:34:28.234          # Ensure that the logcollector is connected to the socket
00:34:28.234          callback_message = lg.callback_socket_connected(socket_name=config['socket_name'],
00:34:28.234                                                          socket_path=config['socket_path'])
00:34:28.234          wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
00:34:28.234                                  error_message=lg.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING,
00:34:28.234  >                               callback=callback_message)
00:34:28.234  
00:34:28.234  test_logcollector/test_location_custom_sockets/test_location_custom_sockets.py:241: 
00:34:28.234  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
00:34:28.234  /usr/local/lib/python3.6/dist-packages/wazuh_testing-4.3.0-py3.6.egg/wazuh_testing/tools/monitoring.py:196: in start
00:34:28.234      error_message=error_message).result()
00:34:28.234  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
00:34:28.234  
00:34:28.234  self = <wazuh_testing.tools.monitoring.QueueMonitor object at 0x7f4bd04a9b38>
00:34:28.234  timeout = 5
00:34:28.234  callback = <function make_callback.<locals>.<lambda> at 0x7f4bd17f7400>
00:34:28.234  accum_results = 1, update_position = True, timeout_extra = 0
00:34:28.234  error_message = 'The expected command monitoring log has not been produced'
00:34:28.234  
00:34:28.234      def start(self, timeout=-1, callback=_callback_default, accum_results=1, update_position=True, timeout_extra=0,
00:34:28.234                error_message=''):
00:34:28.234          """Start the queue monitoring until the stop method is called."""
00:34:28.234          if not self._continue:
00:34:28.234              self._continue = True
00:34:28.234              self._abort = False
00:34:28.234              result = None
00:34:28.234      
00:34:28.234              while self._continue:
00:34:28.234                  if self._abort:
00:34:28.234                      self.stop()
00:34:28.234                      if error_message:
00:34:28.234                          logger.error(error_message)
00:34:28.234                          logger.error(f"Results accumulated: "
00:34:28.234                                       f"{len(result) if isinstance(result, list) else 0}")
00:34:28.234                          logger.error(f"Results expected: {accum_results}")
00:34:28.234  >                   raise TimeoutError(error_message)
00:34:28.234  E                   TimeoutError: The expected command monitoring log has not been produced
00:34:28.234  
00:34:28.234  /usr/local/lib/python3.6/dist-packages/wazuh_testing-4.3.0-py3.6.egg/wazuh_testing/tools/monitoring.py:456: TimeoutError
00:34:28.234  ---------------------------- Captured stderr setup -----------------------------
00:34:28.234  2021/06/23 08:15:26 wazuh-logcollector[21194] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
00:34:28.234  2021/06/23 08:15:26 wazuh-logcollector[21194] main.c:126 at main(): DEBUG: Wazuh home directory: /var/ossec
00:34:28.234  2021/06/23 08:15:26 wazuh-logcollector[21194] agent_op.c:200 at os_read_agent_profile(): DEBUG: Calling os_read_agent_profile().
00:34:28.234  2021/06/23 08:15:26 wazuh-logcollector[21194] agent_op.c:219 at os_read_agent_profile(): DEBUG: os_read_agent_profile() = [centos, centos7, centos7.4]
00:34:28.234  2021/06/23 08:15:26 wazuh-logcollector[21194] config.c:385 at ReadConfig(): DEBUG: agent_config element does not have any attributes.
00:34:28.234  ----------------------------- Captured stderr call -----------------------------
00:34:28.234  2021-06-23 08:15:32,209 - wazuh_testing - ERROR - The expected command monitoring log has not been produced
00:34:28.234  2021-06-23 08:15:32,209 - wazuh_testing - ERROR - Results accumulated: 0
00:34:28.234  2021-06-23 08:15:32,209 - wazuh_testing - ERROR - Results expected: 1
00:34:28.234  ------------------------------ Captured log call -------------------------------
00:34:28.234  ERROR    wazuh_testing:monitoring.py:452 The expected command monitoring log has not been produced
00:34:28.234  ERROR    wazuh_testing:monitoring.py:453 Results accumulated: 0
00:34:28.234  ERROR    wazuh_testing:monitoring.py:455 Results expected: 1
00:34:28.234  - generated html file: file:///tmp/Test_integration_B776_20210623084127/report.html -
00:34:28.234  =========================== short test summary info ============================
00:34:28.234  FAILED test_logcollector/test_location_custom_sockets/test_location_custom_sockets.py::test_location_custom_sockets_offline[target_custom_socket_mode_tcp-batch_500]
00:34:28.234  !!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
00:34:28.234  ====== 1 failed, 475 passed, 12 skipped, 9 xfailed in 1389.77s (0:23:09) =======
00:34:28.234  
```

https://devel.ci.wazuh.info/job/Test_integration/774/console

--- a/deps/wazuh_testing/wazuh_testing/tools/services.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/services.py
@@ -120,7 +120,7 @@ def control_service(action, daemon=None, debug_mode=False):
 
                     for proc in alive:
                         proc.kill()
-                except psutil.NoSuchProcess:
+                except (psutil.NoSuchProcess, psutil.AccessDenied):
                     pass
 
                 delete_sockets(WAZUH_SOCKETS[daemon])

